### DISCUSSION TITLE
What's the best way to use URQL in server components?

### DISCUSSION THREAD

**Discussion by focux** [urql-graphql/urql#2849] [Q&A]
I'm using Next.js 13 with server components and I'm trying to find the best way to use URQL with it. I'm using `urql/core` to query and doing mutations using `client.query` and `client.mutation` respectively. It works well for some cases but I don't like that it doesn't update the data after executing a mutation. You need a full-page reload to get updated values.

Would be good if there were a way to keep the cache capabilities of URQL in server components. Any ideas?

**Comment by JoviDeCroock** [comment]
Instead of using the promisified stuff you can do 

```js
import { subscribe, pipe } from 'wonka'
pipe(client.query(), subscribe(() => { somecallback }))
```

That's how the React components work under the hood as well

**Reply by focux** [reply]
I still don't know how that would help me in my case, do you have an example that you can point me out?

I tested it on server components and there are a few things that don't work, like, for rendering I'd need to await the result, and also, after the first result, I don't get more updates after executing a mutation—I'm using document caching btw.

Thanks for your answer.
